### PR TITLE
Added an AssetStore for handling CRUD operations

### PR DESCRIPTION
**Pull request by ehgoodenough**
## Purpose

As discussed before, the operations for adding and deleting and updating an asset needed to be refactored into a different method, since I wanted both the HTTP (express.js) and TCP (socket.io) to be able to access those operations. I also finally got around to adding the transcoding after the downloading.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,24 +1,68 @@
...
+////////////
 
-    console.log("Example app listening at http://%s:%s", host, port)
+io = socketio(server)
+io.on("connection", function(socket)
+{
+    AssetStore.forEachAsset(function(asset)
+    {
+        socket.emit("add asset", asset)
+    })
```
Comment (tmjz402) [inline server.js:45]: I don't really understand how this works.

Comment (ehgoodenough) [inline server.js:45]: Yeah! We use this to communicate to the web client frontend using Socket.IO, like this: https://github.com/viditor/chrono/blob/master/source/index.js

Comment (ehgoodenough) [inline server.js:45]: For more about socket.io, check out http://socket.io. :]

Comment (tmjz402) [inline server.js:45]: This needs tests.

**Pull request merged**